fakenect/fakenect.c

```diff
@@ -325,9 +436,75 @@ int freenect_open_device_by_camera_serial(freenect_context *ctx, freenect_device
     return 0;
 }
 
+static void read_device_info(freenect_device *dev)
+{
+	char fn[512];
+	snprintf(fn, sizeof(fn), "%s/device.json", input_path);
+
+	JSON_Value *js = json_parse_file(fn);
```

Comment:
Should it skip all this if `device.json` doesn't exist?  Ideally old recordings will still work.

Reply:
Yeah makes sense. In that case it probably also makes sense to move the follow-up call to `freenect_init_registration` to the bottom of `read_device_info` at a point where we know we have read the required `reg_info` state.

I've tested an updated version with this change and will update my pull request.

Since I decided to return silently if we fail to load a device.json (for compatibility) I also added a check in `freenect_set_depth_mode` whereby requesting `_DEPTH_MM` will result in a *"Warning:..."* if registration state wasn't previously loaded.